clients/drcachesim/common/memtrace_stream.h

```diff
@@ -155,10 +155,25 @@ class memtrace_stream_t {
         return false;
     }
 
+    /**
+     * Returns the 0-based ordinal for the current shard.  For parallel analysis,
+     * this equals the \p shard_index passed to parallel_shard_init_stream().
+     * This is more useful for serial modes where there is no other convenience mechanism
+     * to determine such an index; there, it equals the same index that would be used in
+     * parallel mode, allowing a tool to compute per-shard results even in serial mode. If
+     * not implemented, -1 is returned.
+     */
+    virtual int
+    get_shard_index() const
+    {
+        return -1;
+    }
+
     /**
      * Returns a unique identifier for the current "output cpu".  Generally this only
```

Comment:
You mean fixed for an output stream?  The result of get_output_cpu is always fixed for that output stream.  For as-traced the cpu id comes from the input stream and can vary across one input stream (since it's a software thread that might have run on many cpus), but it won't vary on the output stream.